src/qibo/models/dbi/double_bracket.py

```diff
@@ -156,6 +203,98 @@ def hyperopt_step(
         )
         return best["step"]
 
+    def polynomial_step(
+        self,
+        n: int = 4,
+        n_max: int = 5,
+        d: np.array = None,
+        backup_scheduling: DoubleBracketScheduling = None,
+    ):
+        r"""
+        Optimizes iteration step by solving the n_th order polynomial expansion of the loss function.
+        e.g. $n=2$: $2\Trace(\sigma(\Gamma_1 + s\Gamma_2 + s^2/2\Gamma_3)\sigma(\Gamma_0 + s\Gamma_1 + s^2/2\Gamma_2))
+        Args:
+            n (int, optional): the order to which the loss function is expanded. Defaults to 4.
+            n_max (int, optional): maximum order allowed for recurring calls of `polynomial_step`. Defaults to 5.
+            d (np.array, optional): diagonal operator, default as $\delta(H)$.
+            backup_scheduling (`DoubleBracketScheduling`): the scheduling method to use in case no real positive roots are found.
+        """
+
+        if d is None:
+            d = self.diagonal_h_matrix
+
+        if backup_scheduling is None:
+            backup_scheduling = DoubleBracketScheduling.grid_search
+
+        def sigma(h: np.array):
+            return h - self.backend.cast(np.diag(np.diag(self.backend.to_numpy(h))))
+
+        def Gamma(k: int):
+            r"""Computes the k_th Gamma function i.e $\Gamma_k=[W,...,[W,[W,H]]...]$, where we take k nested commutators with $W = [D, H]$"""
+            if k == 0:
+                return self.h.matrix
+            else:
+                W = self.commutator(d, sigma(self.h.matrix))
+                result = self.h.matrix
+                for _ in range(k):
+                    result = self.commutator(W, result)
+            return result
+
+        # list starting from s^n highest order to s^0
+        sigma_gamma_list = np.array([sigma(Gamma(k)) for k in range(n + 2)])
```

Comment:
The evaluation of Gammas seems time consuming,  using an iterative approach could save some time, since:
$\Gamma_k = [W,\Gamma_{k-1}]$
with $\Gamma_0 = H$